androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LauncherActivity.java

```diff
@@ -172,6 +179,30 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         new TwaSharedPreferencesManager(this)
                 .writeLastLaunchedProviderPackageName(mTwaLauncher.getProviderPackage());
+
+        mPackageManager = getPackageManager();
+        // Check whether ManageDataLauncherActivity is present in manifest.
+        Intent siteSettingsIntent = new Intent(this, ManageDataLauncherActivity.class);
+        siteSettingsIntent.setAction(ManageDataLauncherActivity.ACTION_MANAGE_TRUSTED_WEB_ACTIVITY_DATA);
+        List<ResolveInfo> activities = mPackageManager.queryIntentActivities(siteSettingsIntent,
+                PackageManager.MATCH_DEFAULT_ONLY);
+        boolean siteSettingsAvailable = activities.size() > 0;
+
+        // Check whether TWA provider package supports site settings.
+        String packageName = mTwaLauncher.getProviderPackage();
+        boolean supportsTWASiteSettings = ManageDataLauncherActivity.packageSupportsSiteSettings(packageName, mPackageManager);
+
+        if(siteSettingsAvailable && supportsTWASiteSettings) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
+                ShortcutManager shortcutManager = getSystemService(ShortcutManager.class);
+                ShortcutInfo shortcut = new ShortcutInfo.Builder(this, "siteSettings")
+                        .setShortLabel("Site Settings")
+                        .setLongLabel("Manage website notifications, permissions, etc.")
+                        .setIntent(siteSettingsIntent)
+                        .build();
+                shortcutManager.setDynamicShortcuts(Collections.singletonList(shortcut));
+            }
+        }
```

Comment:
It's a much smaller case, but if we previously had a shortcut and now our browser no-longer supports them (eg, if the user changed their default browser), we should remove the old shortcut.